### DISCUSSION TITLE
is there any way to create dynamic route after /

### DISCUSSION THREAD

**Discussion by rizwan92** [vercel/next.js#11677] [Help]
# is there any way to create a dynamic route after /

 I have created dynamic route inside page directory like below 

- pages
           - [username].js

 When someone visits http://mydomain.com/myusername.
 I get the username from get initial props and show them relevant content 
 problem Is when I visit my other routes like /home/preview or /contacts it gives me the right page but [username].js page run as well I have some code that should only run for [username].js not for another route

routes structure

- - pages

      - contact.js
      - inex.js
      - [username].js
      - home
           - preview.js

we cannot create dynamic routes for  after root 

## To Reproduce
Create a  directory structure like above
1. create the file and create react component for each page
2. console log the `query. username` inside get initial props of [username].js
3. visit `localhost:3000/myusername `, you will see the log of `my username` there 
4. now visite another route like `localhost:3000/contact` it will result in you your contacts.js react component but it will also run the [username].js file
5 likely you can also visit /home/preview and it will again run the file 

## Expected behavior

it should not run the [username].js file

**Comment by timneutkens** [comment]
Tried locally and it works exactly as expected:

- `/contact` routes to `pages/contact.js`
- `/home/preview` routes to `pages/home/preview.js`
- `/` routes to `pages/index.js`
- `/test12345` routes to `pages/[username].js`

<img width="375" alt="Screen Shot 2020-04-05 at 13 34 43" src="https://user-images.githubusercontent.com/6324199/78473631-72746900-7742-11ea-8ee7-7ef843a1363e.png">

Can you provide a full repository?

**Comment by rizwan92** [comment]
@timneutkens  you are right i also tried it creating new next project and test it I am not able to get this problem 

but believe me, I am facing this issue in my project I don't know why I can show you this in video

**Comment by timneutkens** [comment]
As asked for please provide a complete reproduction so that we can investigate.

**Comment by rizwan92** [comment]
I Created the reproduction now you visit any URL it will run the [username].js file

[github repositoty](https://github.com/rizwan92/next-bug)

**Comment by timneutkens** [comment]
@rizwan92 Thanks! Can you make it public so that I can have a look

**Comment by rizwan92** [comment]
ohh my bad it is public now

**Comment by timneutkens** [comment]
Just tried the repository, it works correctly, same as my earlier reply:

- `/contact` routes to `pages/contact.js`
- `/home/preview` routes to `pages/home/preview.js`
- `/` routes to `pages/index.js`
- `/test12345` routes to `pages/[username].js`

**Comment by rizwan92** [comment]
watch this video [movie](https://res.cloudinary.com/relive-cloud/video/upload/v1586240600/digitalcard/Screen_Recording_2020-04-07_at_11.46.51_AM_sqyfal.mov)


and why getInitialProps run Twice in server-side
` console.log(typeof windows === "undefined") 
`
it will log true twice
